core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/TrafParallelClientScanner.java.tmpl

```diff
@@ -116,21 +116,22 @@ public class TrafParallelClientScanner extends AbstractClientScanner implements
     LOG.debug("Found "+locs.size()+" region(s).");
     if (locs.size() == 0) return;
 
+    int threads = parallelScaling>1.0 ?(int)Math.ceil(parallelScaling) : (int)Math.ceil(locs.size() * parallelScaling);
+
```

Comment:
there is 2 ways to set DOP: as a % of regions to scan, or define the number of thread manually.
parallelScaling = 0 disabled.
0<parallelScaling<=1.0, DOP expressed as percentage of regions to scan. 1.0 means 100%, .2 means 20%,
parallelScanner > 1.0, parallelScanner should be an integer number representing the number of thread you want. If you define parallelScanner > 1 but not integer, we convert using ceiling...